graphicsDatatypes.h

```diff
@@ -0,0 +1,34 @@
+#ifndef GRAPHICSDATATYPES_H_
+#define GRAPHICSDATATYPES_H_
+
+typedef struct{
+    uint8_t reserved;
+
+    uint8_t redValue;
+    uint8_t greenValue;
+    uint8_t blueValue;
+}colorStruct;
+
+typedef union{
+    uint32_t colorValue;
+    colorStruct colorSeperate;
+}color;
+
+typedef struct{
+    uint16_t xPosition;
+    color curserColor;
+    uint8_t curserVisible;
+}curserStruct;
+
+/*position is defined in the top left pixel of the character*/
+typedef struct{
+    uint16_t xPosition;
+    uint16_t yPosition;
+
+    color characterColor;
+
+    uint8_t characterPixel[5];
+
+}character;
```

Comment:
- Textdarstellung in eigenes Text-Modul:
Der Text wird ganz normal als char* festgelegt. Zu jedem ASCII-Zeichen wird der Glyph aus einer Lookup-Tabelle ausgelesen, und dann pixelweise an die angegebene Stelle gezeichnet.
- "Textfelder" (mit X/Y Koordinaten und Farbe) in UI-Modu

Reply:
Text ist in eigenem Modul s.o.. Kommt später ins Repository. Mein Ansatz ähnelt dem in deinem Kommentar, sage aber erst später etwas dazu. Zeichen-Datenbank würde ich gerne als Array Makros machen oder wie macht man das am besten?

Reply:
In text.c ein `uint8_t aucTextGlyphLUT[Anzahl_Zeichen_Im_Zeichensatz][Zeichenhöhe_Pixel]` anlegen, wo man den Glyph per `aucGlyph = aucTextGlyphLUT['ASCII-Buchstabe'-Zeichen_Offset]` auswählen kann. Die Glyphen werden dann Zeilenweise über `aucGlyph[Zeilennummer]`, und Spaltenweise über die Bitstelle (Bit 5 = linker Zeichenrand) gezeichnet (siehe Konzept-Dokument).
Die ganzen Glyphen trägt man als Initialisierungswerte für die LUT ein (LUT am Besten auch const machen).